docs/src/manual/models.md

```diff
@@ -62,50 +62,66 @@ functions.
 
 Consult the docstsrings of each attribute for information on what it represents.
 
-## Model attributes
+## ModelLike API
 
-The following attributes are available for models:
+Objects subtyping [`ModelLike`](ref) must implement:
+
+ * [`is_empty`](@ref)
+ * [`empty!`](@ref)
+
+The following attributes are available. Some are required by all
+implementations, and others are optional.
+
+**Required**
+
+ * [`ListOfConstraintAttributesSet`](@ref)
+ * [`ListOfConstraintIndices`](@ref)
+ * [`ListOfConstraintTypesPresent`](@ref)
+ * [`ListOfModelAttributesSet`](@ref)
+ * [`ListOfVariableAttributesSet`](@ref)
+ * [`ListOfVariableIndices`](@ref)
+ * [`NumberOfConstraints`](@ref)
+ * [`NumberOfVariables`](@ref)
+
+**Optional**
 
  * [`Name`](@ref)
  * [`ObjectiveFunction`](@ref)
  * [`ObjectiveFunctionType`](@ref)
  * [`ObjectiveSense`](@ref)
- * [`NumberOfVariables`](@ref)
- * [`ListOfVariableIndices`](@ref)
- * [`ListOfConstraintTypesPresent`](@ref)
- * [`NumberOfConstraints`](@ref)
- * [`ListOfConstraintIndices`](@ref)
- * [`ListOfOptimizerAttributesSet`](@ref)
- * [`ListOfModelAttributesSet`](@ref)
- * [`ListOfVariableAttributesSet`](@ref)
- * [`ListOfConstraintAttributesSet`](@ref)
 
-## Optimizer attributes
-
-The following attributes are available for optimizers:
-
- - [`SolverName`](@ref)
- - [`RawOptimizerAttribute`](@ref)
- - [`Silent`](@ref)
- - [`TimeLimitSec`](@ref)
- - [`NumberOfThreads`](@ref)
- - [`RawSolver`](@ref)
-
-In addition, optimizers should implement the following attributes to provide
-access to solutions:
-
- - [`TerminationStatus`](@ref)
- - [`TerminationStatusCode`](@ref)
- - [`PrimalStatus`](@ref)
- - [`DualStatus`](@ref)
- - [`ResultStatusCode`](@ref)
- - [`RawStatusString`](@ref)
- - [`ResultCount`](@ref)
- - [`ObjectiveValue`](@ref)
- - [`DualObjectiveValue`](@ref)
- - [`ObjectiveBound`](@ref)
- - [`RelativeGap`](@ref)
- - [`SolveTimeSec`](@ref)
- - [`SimplexIterations`](@ref)
- - [`BarrierIterations`](@ref)
- - [`NodeCount`](@ref)
+## AbstractOptimizer API
+
+In addition to implementing the [ModelLike API](@ref), objects subtyping
```

Comment:
I woulf not say that solvers have to implement the required attributes above. One may choose that the solver cannot be copied from for instance. It makes sense to what need to be implemented in order to be copied from but a solver is allowed to only implement `copy_to_and_optimize!`, the empty functions and getters for the results and nothing else